src/main/resources/com/atlassian/jira/cloud/jenkins/configuration/config.jelly

```diff
@@ -1,8 +1,158 @@
 <?jelly escape-by-default='true'?>
-<j:jelly xmlns:j="jelly:core" xmlns:l="/lib/layout" >
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:l="/lib/layout" xmlns:f="/lib/form" xmlns:c="/lib/credentials">
+
     <l:layout permission="${app.ADMINISTER}" title="${%title}" norefresh="true" type="one-column">
         <l:main-panel>
+            <link rel="stylesheet" type="text/css" href="${resURL}/plugin/atlassian-jira-software-cloud/config.css"/>
+            <script src="${resURL}/plugin/atlassian-jira-software-cloud/config.js"/>
+            <script src="${resURL}/plugin/atlassian-jira-software-cloud/regexTester.js"/>
+
             <h1>Atlassian Software Cloud</h1>
+            <p>Send build and deployment data to connected Jira Software Cloud sites.</p>
+            <div>
+                <st:if test="${config != null}">
+                    <f:form method="post" action="saveConfiguration" id="saveConfigurationForm">
+                        <!-- EXISTING SITES -->
+                        <div class="margin-bottom-2em">
+                            <f:section title="Connected Sites">
+                                <table class="jenkins-table sortable jenkins-!-margin-bottom-0 site-table">
+                                    <thead>
+                                        <tr>
+                                            <th class="col-25">Site Name</th>
+                                            <th class="col-35">Webhook URL</th>
+                                            <th class="col-10">Actions</th>
+                                        </tr>
+                                    </thead>
+                                    <tbody>
+                                        <j:set var="sites" value="${config.getSites()}" />
+                                        <j:forEach var="site" items="${sites}" varStatus="loop">
+                                            <!-- hidden values to assign to JSON tree structure-->
```

Comment:
Jenkins does a lot of things its own way, that includes JSON from a form, its converts it into a tree like structure, with that we had to get a bit cute with how the values are displayed, saves manipulated on the backend, but we have the actual values on a hidden form so we can pretend to be normal